kernel/private/classes/exceptions/database/exception.php

```diff
@@ -17,5 +17,30 @@
  */
 class eZDBException extends ezcBaseException
 {
+    /**
+     * Original message, before escaping
+     */
+    public $originalMessage;
+
+    /**
+     * Constructs a new eZDBException with $message and $code
+     *
+     * @param string $message
+     * @param int $code
+     */
+    public function __construct( $message, $code = 0 )
+    {
+        $this->originalMessage = $message;
+        $this->code = $code;
+
+        if ( php_sapi_name() == 'cli' )
+        {
+            $this->message = $message;
+        }
+        else
+        {
+            $this->message = htmlspecialchars( $message, ENT_QUOTES, 'UTF-8' );
```

Comment:
Nitpick: it is bad practice to html-encode data on data-creation. Data should be escaped when it is displayed.